datanode/gateway/graphql/resolvers.go

```diff
@@ -855,6 +855,161 @@ func (r *myQueryResolver) EstimateOrder(
+	if price != nil {
+		order.Price = *price
+	}
+	s, err := safeStringUint64(size)
+	if err != nil {
+		return nil, err
+	}
+	order.Size = s
+	if len(market) <= 0 {
+		return nil, errors.New("market missing or empty")
+	}
+	order.MarketId = market
+	if len(party) <= 0 {
+		return nil, errors.New("party missing or empty")
+	}
+
+	order.PartyId = party
+	order.TimeInForce = timeInForce
+	order.Side = side
+	order.Type = ty
+
+	// GTT must have an expiration value
+	if order.TimeInForce == types.Order_TIME_IN_FORCE_GTT && expiration != nil {
+		order.ExpiresAt = vegatime.UnixNano(*expiration).UnixNano()
+	}
+
+	req := v2.EstimateFeeRequest{
+		MarketId: order.MarketId,
+		Price:    order.Price,
+		Size:     order.Size,
+	}
+
+	// Pass the order over for consensus (service layer will use RPC client internally and handle errors etc)
+	resp, err := r.tradingDataClientV2.EstimateFee(ctx, &req)
+	if err != nil {
+		r.log.Error("Failed to get fee estimates using rpc client in graphQL resolver", logging.Error(err))
+		return nil, err
+	}
+
+	// calclate the fee total amount
+	var mfee, ifee, lfee num.Decimal
+	// errors doesn't matter here, they just give us zero values anyway for the decimals
+	if len(resp.Fee.MakerFee) > 0 {
+		mfee, _ = num.DecimalFromString(resp.Fee.MakerFee)
+	}
+	if len(resp.Fee.InfrastructureFee) > 0 {
+		ifee, _ = num.DecimalFromString(resp.Fee.InfrastructureFee)
+	}
+	if len(resp.Fee.LiquidityFee) > 0 {
+		lfee, _ = num.DecimalFromString(resp.Fee.LiquidityFee)
+	}
+
+	fees := &TradeFee{
+		MakerFee:          resp.Fee.MakerFee,
+		InfrastructureFee: resp.Fee.InfrastructureFee,
+		LiquidityFee:      resp.Fee.LiquidityFee,
+	}
+
+	return &FeeEstimate{
+		Fees:           fees,
+		TotalFeeAmount: decimal.Sum(mfee, ifee, lfee).String(),
+	}, nil
+}
+
+func (r *myQueryResolver) EstimatePosition(
+	ctx context.Context,
+	marketId string,
+	openVolume string,
+	orders []*OrderInfo,
+	collateralAvailable *string,
+) (*PositionEstimate, error) {
+	collat := ""
+	if collateralAvailable != nil {
+		collat = *collateralAvailable
+	}
+
+	ov, err := safeStringInt64(openVolume)
+	if err != nil {
+		return nil, err
+	}
+
+	ord := make([]*v2.OrderInfo, 0, len(orders))
+	for _, o := range orders {
+		mktOrder := false
+		if o.IsMarketOrder != nil {
+			mktOrder = *o.IsMarketOrder
+		}
+
+		r, err := safeStringUint64(o.Remaining)
+		if err != nil {
+			return nil, err
+		}
+
+		ord = append(ord, &v2.OrderInfo{
+			Side:          o.Side,
+			Price:         o.Price,
+			Remaining:     r,
+			IsMarketOrder: mktOrder,
+		})
+	}
+
+	req := &v2.EstimatePositionRequest{
+		MarketId:            marketId,
+		OpenVolume:          ov,
+		Orders:              ord,
+		CollateralAvailable: collat,
+	}
+
+	resp, err := r.tradingDataClientV2.EstimatePosition(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var liquidation *LiquidationEstimate
+	if resp.Liquidation != nil {
+		worst := &LiquidationPrice{
```

Comment:
Could you set the bindings to the proto types in the graphql config so we can avoid doing those mappings here?

Reply:
Done now.